backend/src/main/java/reviewme/template/service/TemplateMapper.java

```diff
@@ -0,0 +1,100 @@
+package reviewme.template.service;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import reviewme.question.domain.OptionGroup;
+import reviewme.question.domain.OptionItem;
+import reviewme.question.domain.Question2;
+import reviewme.question.domain.exception.NoOptionItemsInOptionGroupException;
+import reviewme.question.repository.OptionGroupRepository;
+import reviewme.question.repository.OptionItemRepository;
+import reviewme.question.repository.Question2Repository;
+import reviewme.reviewgroup.domain.ReviewGroup;
+import reviewme.template.domain.Section;
+import reviewme.template.domain.Template;
+import reviewme.template.dto.response.OptionGroupResponse;
+import reviewme.template.dto.response.OptionItemResponse;
+import reviewme.template.dto.response.QuestionResponse;
+import reviewme.template.dto.response.SectionResponse;
+import reviewme.template.dto.response.TemplateResponse;
+import reviewme.template.repository.SectionRepository;
+
+@Component
+@RequiredArgsConstructor
+public class TemplateMapper {
+
+    private final SectionRepository sectionRepository;
+    private final Question2Repository questionRepository;
+    private final OptionGroupRepository optionGroupRepository;
+    private final OptionItemRepository optionItemRepository;
+
+    public TemplateResponse mapToTemplateResponse(ReviewGroup reviewGroup, Template template) {
+        List<SectionResponse> sectionResponses = template.getSectionIds()
+                .stream()
+                .map(this::mapToSectionResponse)
+                .toList();
+
+        return new TemplateResponse(
+                template.getId(),
+                reviewGroup.getReviewee(),
+                reviewGroup.getProjectName(),
+                sectionResponses
+        );
+    }
+
+    private SectionResponse mapToSectionResponse(long sectionId) {
+        Section section = sectionRepository.getSectionById(sectionId);
+        List<QuestionResponse> questionResponses = section.getQuestionIds()
+                .stream()
+                .map(this::mapToQuestionResponse)
+                .toList();
+
+        return new SectionResponse(
+                sectionId,
+                section.getVisibleType().name(),
+                section.getOnSelectedOptionId(),
+                section.getHeader(),
+                questionResponses
+        );
+    }
+
+    private QuestionResponse mapToQuestionResponse(long questionId) {
+        Question2 question = questionRepository.getQuestionById(questionId);
+        OptionGroupResponse optionGroupResponse = optionGroupRepository.findByQuestionId(question.getId())
+                .map(this::mapToOptionGroupResponse)
+                .orElse(null);
+
+        return new QuestionResponse(
+                question.getId(),
+                question.isRequired(),
+                question.getContent(),
+                question.getContent(),
+                optionGroupResponse,
```

Comment:
Optional로 null 가능성을 명시하면서 안전성을 챙기는 건 어떨까요?
```suggestion
         Optional<OptionGroupResponse> optionGroupResponse = optionGroupRepository.findByQuestionId(question.getId())
                .map(this::mapToOptionGroupResponse);

        return new QuestionResponse(
                question.getId(),
                question.isRequired(),
                question.getContent(),
                question.getContent(),
                optionGroupResponse.orElse(null),
        );
```